.github/workflows/nightly.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish Nightly Builds
+
+on:
+  push:
+    branches:
+      - nightly_build
+
+jobs:
+  build-test-publish:
+    strategy:
+      matrix:
+        include:
+          - runson: macos-latest # mac
+            archive_postfix: macos-arm64
+
+          - runson: ubuntu-latest # linux x64
+            image: docker://public.ecr.aws/async-profiler/asprof-builder-x86:latest
+            arch: x86_64
+            archive_postfix: linux-x64
+
+          - runson: codebuild-test-${{ github.run_id }}-${{ github.run_attempt }} # linux arm
+            arch: aarch64
+            archive_postfix: linux-arm64
+    runs-on: ${{ matrix.runson }}
+    container:
+      image: ${{ matrix.image }} # will be null except for linux x64
+      options: --privileged # needed in order to use sysctl
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build and Test
+        run: |
+          if [[ "${{ matrix.runson }}" != "macos-latest" ]]; then # configuration needed only for linux
+            sudo sysctl kernel.perf_event_paranoid=1 && sudo sysctl kernel.kptr_restrict=0 # allow tests to execute
+            CC=/usr/local/musl/${{ matrix.arch }}/bin/musl-gcc
+          fi
+          git config --global --add safe.directory $PWD # needed to avoid `fatal: detected dubious ownership in repository at '/__w/async-profiler/async-profiler'` error
+          make test # build and test
+          make clean # remove test files
+          make # build again without test files
+      - uses: actions/upload-artifact@v4
+        with:
+          name: async-profiler-nightly-${{ github.sha }}-${{ matrix.archive_postfix }}
+          # Uploading to the workflow automatically zips up the files, so we should upload the build directory instead of an already zipped release.
+          # https://github.com/actions/upload-artifact?tab=readme-ov-file#zip-archives
+          path: build/
```

Comment:
I don't see where `release` target is called.
Contents of `/build` is not the same as the contents of the release archive.

Reply:
fixed this